### PR TITLE
MM-47489 Assume Remote Cluster Service enabled if Shared Channels enabled.

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
The ticket linked below intended to enable Shared Channels for Professional licenses.  The PRs missed adding the dependency Remote Cluster Service to the professional license.  This PR corrects that by assuming Remote Cluster Service is enabled if Shared Channels is enabled.  It also simplifies configuration by allowing users to simply set the Shared Channels config item.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-47489

#### Release Note
```release-note
Shared Channels feature included in Professional license.
```

### CODE REVIEW

File: app/server.go [hunk 1]
```diff
@@ -579,8 +579,8 @@ func (s *Server) startInterClusterServices(license *model.License) error {
 
 	// Remote Cluster service
 
-	// License check
-	if !*license.Features.RemoteClusterService {
+	// License check  (assumed enabled if shared channels enabled)
```
Comment (neallred) [inline app/server.go:582]: Super nit: I think there is an extra space between `check` and the parenthesis.

**Review by neallred — APPROVED** [review]
LGTM, thanks @wiggin77 !

File: app/server.go [hunk 2]
```diff
@@ -579,14 +579,14 @@ func (s *Server) startInterClusterServices(license *model.License) error {
...
```
Comment (wiggin77) [inline app/server.go:582]:
```suggestion
	// License check (assume enabled if shared channels enabled)
```

File: app/server.go (hunk 1 above)
Comment (wiggin77) [reply app/server.go]: Fixed. Thanks.

**Review by esethna — APPROVED** [review]
@amyblais please backport to supported releases v7.7+

**Comment by amyblais** [comment]
@wiggin77 Since Vishal is out of office, maybe another dev can be assigned to review this?

**Review by AshishDhama — APPROVED** [review]

**Comment by amyblais** [comment]
@wiggin77 Please help open manual cherry-pick PRs for ``cloud``, ``release-7.8``, ``release-7.7`` branches.

**Pull request merged**